src/EditorFeatures/CSharpTest/CommentSelection/ToggleBlockCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,1993 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.CSharp.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Utilities;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Experiments;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CommentSelection
+{
+    [UseExportProvider]
+    public class ToggleBlockCommentCommandHandlerTests
+    {
+        private const string BlockCommentOpenMarker = "/*";
+        private const string BlockCommentCloseMarker = "*/";
+
+        private class MockCommentSelectionService : AbstractCommentSelectionService
+        {
+            public MockCommentSelectionService()
+            {
+                SupportsBlockComment = true;
+            }
+
+            public override string SingleLineCommentString => "//";
+            public override bool SupportsBlockComment { get; }
+            public override string BlockCommentStartString => BlockCommentOpenMarker;
+            public override string BlockCommentEndString => BlockCommentCloseMarker;
+        }
```

Comment:
Why do we need a mock here?  Why not just actually use the C# selection service?